### PR TITLE
Add 'services' as a root element in docker-compose.yml #306

### LINKED ISSUE #306: Add  'services' as a root element in docker-compose.yml
## Describe the Enhancement ##
*Starting from Compose specification v2 compose files using the version 2 syntax must indicate the version number at the root of the document. All services must be declared under the services key. Compose specification v1 is deprecated. We need to update  docker-compose.yml to be compatible with v3 by adding 'services' as a root element 
https://docs.docker.com/compose/compose-file/compose-versioning/#version-3
* 

### Time Estimate ###
*Estimate the amount of work required here.*
*Issues should represent approximately 1h of work.*

### Sub-Issues ###
Consider breaking the enhancement down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
*List relevant project deadlines here or state NONE.*

### Funding Source ###
*Define the source of funding and account keys here or state NONE.*

### Assignee ###
- [ ] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [ ] Select **component(s)**
- [ ] Select **priority**
- [ ] Select **requestor(s)**

### Projects and Milestone ###
- [ ] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED PROJECT ASSIGNMENT** label
- [ ] Select **Milestone** as the next official version or **Future Versions**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [ ] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Enhancement Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Linked issues**
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by TatianaBurek**
## Pull Request Testing ##

- [X] Describe testing already performed for these changes:
 Tested with Docker 20.10.7 and Compose 1.29.2

- [X] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:
- Test with earlier version of Docker and Compose

- [X] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[No]**

- [X] Do these changes include sufficient testing updates? **[No]**

- [X] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the PR definition above.
- [ ] Ensure the PR title matches the feature or bugfix branch name.
- [ ] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [ ] After submitting the PR, select **Linked issues** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -1,29 +1,31 @@
-db:
-    image: mysql:5.7
-    container_name: mysql_mv
-    user: "mysql:mysql"
-    ports:
-        - "6603:3306"
-    volumes:
-        - ${MYSQL_DIR}:/var/lib/mysql
+version: "3.0"
```
Comment (JohnHalleyGotway) [inline docker/docker-compose.yml:1]: Tatiana, I'm curious about this "version" entry in the docker-compose.yml file. Is "3.0" meant to indicate the METviewer software version number... or the version number of this docker-compose.yml file? The version of the METviewer software is actually controlled by ${METVIEWER_DOCKER_IMAGE}. If we're introducing a new level of versioning, we'd need documentation to clarify what it actually means.

Sorry for the delay on this PR. I missed it 8 days ago.

Comment (TatianaBurek) [reply docker/docker-compose.yml:1]: "version" is a Compose file version with is supported by Compose 1.13.0+

Comment (JohnHalleyGotway) [reply docker/docker-compose.yml:1]: Ah OK, thanks for clarifying. I suspect that this describes that v3 format:
https://docs.docker.com/compose/compose-file/compose-file-v3/

This seems fine then. I'll proceed with testing on my Mac using Docker Desktop version 3.5.1 and Docker Engine 20.10.7. That's what I have available.
https://docs.docker.com/compose/compose-file/compose-file-v3/

**Review by JohnHalleyGotway — APPROVED** [review]
I approve of these changes (and sorry for the long delay!).
I tested on my Mac using Docker Desktop version 3.5.1 with Engine 20.10.7 and Compose 1.29.2.
```
mkdir -p data/mysql data/metviewer data/metviewer_data
export MYSQL_DIR=`pwd`/data/mysql
export METVIEWER_DIR=`pwd`/data/metviewer
export METVIEWER_DATA=`pwd`/data/metviewer_data
export METVIEWER_DOCKER_IMAGE=dtcenter/metviewer
docker-compose up -d
```
And that all ran without error.
Also note that the following runs fine as well:
```
docker compose up -d
docker compose down
```

**Pull request merged**